File: LeetCode/Kotlin/s1052.kt
class Solution1052 {
    fun maxSatisfied(customers: IntArray, grumpy: IntArray, minutes: Int): Int {
        var l = 0
        var r = 0
        var maxUnsatisfied = 0
        var unSatisfied = 0
        var satisfied = 0
        while (r<grumpy.size) {
            if (grumpy[r] != 0) unSatisfied += customers[r++] else satisfied += customers[r++]
            if (r-l>minutes) l++ .also {if (grumpy[it] != 0) unSatisfied -= customers[it]}
            maxUnsatisfied = maxOf(maxUnsatisfied, unSatisfied)
        }
        return satisfied + maxUnsatisfied
    }
}

File: LeetCode/Rust/s1052.rs
struct Solution1052 {}

impl Solution1052 {
    pub fn max_satisfied(customers: Vec<i32>, grumpy: Vec<i32>, minutes: i32) -> i32 {
        let (mut l, mut r) = (0, 0);
        let (mut satisfied, mut unsatisfied, mut max_unsatisfied) = (0, 0, 0);
        while r<grumpy.len() {
            if grumpy[r] != 0 { unsatisfied += customers[r] } else { satisfied += customers[r] }
            r += 1;
            if r-l>minutes as usize { if grumpy[l] != 0 {unsatisfied -= customers[l]}; l +=1; }
            if unsatisfied > max_unsatisfied { max_unsatisfied = unsatisfied }
        }
        satisfied + max_unsatisfied
    }

}